# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/tests/test_pasteImgFromTransferable.html
# Commit: d6c4fdcf6a69
# Full Hash: d6c4fdcf6a69ebaa1b133ef8c57da9eef389f89a
# Author: Tom Schuster <evilpies@gmail.com>
# Date: 2022-10-20 21:51:26
# Regressor Bug: 1778565
# File Overlap Count: 1
# Description:
#   Bug 1778565 - Editor: Handle nsIFile transferable as blob. r=nika,masayuki
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155753
# ==============================================================================

diff -r 88976c36da0a -r d6c4fdcf6a69 editor/libeditor/tests/test_pasteImgFromTransferable.html
--- a/editor/libeditor/tests/test_pasteImgFromTransferable.html	Thu Oct 20 06:59:31 2022 -0400
+++ b/editor/libeditor/tests/test_pasteImgFromTransferable.html	Thu Oct 20 11:11:59 2022 +0000
@@ -60,8 +60,13 @@
     trans.init(null);
     trans.setTransferData(test.mimeType, stringStream);
 
+    let evt = new Promise(resolve =>
+      edit.addEventListener("input", resolve, {once: true}));
+
     getHTMLEditor(window).pasteTransferable(trans);
 
+    await evt;
+
     is(edit.innerHTML,
        "<img src=\"data:" + test.mimeType + ";base64," + test.base64 + "\" alt=\"\">",
        "pastedTransferable pastes image as data URL");
